src/grandorgue/model/GOCoupler.cpp

```diff
@@ -326,40 +326,31 @@ void GOCoupler::ChangeKey(int note, unsigned velocity) {
       SetOut(m_CurrentTone + m_Keyshift, m_KeyVelocity[m_CurrentTone]);
 
     if (m_IsNewBasMel) {
-      int nextCandidate = nextNote;
       if (m_CouplerType == COUPLER_BASS) {
-        for (nextCandidate += 1; nextCandidate < (int)m_KeyVelocity.size();
-             nextCandidate++)
-          if (m_KeyVelocity[nextCandidate] > 0)
+        for (nextNote += 1; nextNote < (int)m_KeyVelocity.size(); nextNote++)
```

Comment:
My idea was not only to rename `nextCandidate` to `nextNote` but to remove this code at all because `nextNote` has already been calculated at lines 303-314

Reply:
@oleg68 It's actually not at all about calculating nextNote... it's actually about how to determine/keep m_LastTone.